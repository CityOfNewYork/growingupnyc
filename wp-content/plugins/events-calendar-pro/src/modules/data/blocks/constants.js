//
// ─── BLOCK TYPES ────────────────────────────────────────────────────────────────
//

export const RECURRING = 'recurring';
export const EXCEPTION = 'exception';

export const BLOCK_TYPES = [ RECURRING, EXCEPTION ];

//
// ─── STATE KEYS ─────────────────────────────────────────────────────────────────
//

export const KEY_TYPE = 'type';
export const KEY_ALL_DAY = 'all_day';
export const KEY_MULTI_DAY = 'multi_day';
export const KEY_MULTI_DAY_SPAN = 'multi_day_span';
export const KEY_START_TIME = 'start_time';
export const KEY_END_TIME = 'end_time';
export const KEY_START_TIME_INPUT = '_start_time_input';
export const KEY_END_TIME_INPUT = '_end_time_input';
export const KEY_START_DATE = 'start_date';
export const KEY_START_DATE_INPUT = '_start_date_input';
export const KEY_START_DATE_OBJ = '_start_date_obj';
export const KEY_END_DATE = 'end_date';
export const KEY_END_DATE_INPUT = '_end_date_input';
export const KEY_END_DATE_OBJ = '_end_date_obj';
export const KEY_LIMIT = 'limit';
export const KEY_LIMIT_DATE_INPUT = '_limit_date_input';
export const KEY_LIMIT_DATE_OBJ = '_limit_date_obj';
export const KEY_LIMIT_TYPE = 'limit_type';
export const KEY_BETWEEN = 'between';
export const KEY_DAYS = 'days';
export const KEY_WEEK = 'week';
export const KEY_DAY = 'day';
export const KEY_MONTH = 'month';
export const KEY_TIMEZONE = 'timezone';
