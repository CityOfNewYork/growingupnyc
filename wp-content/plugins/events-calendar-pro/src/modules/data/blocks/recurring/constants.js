/**
 * External dependencies
 */
import { invert } from 'lodash';
import { __ } from '@wordpress/i18n';

//
// ─── RECURRENCE TYPES ───────────────────────────────────────────────────────────
//

export const DAILY = 'daily';
export const WEEKLY = 'weekly';
export const MONTHLY = 'monthly';
export const YEARLY = 'yearly';
export const SINGLE = 'single';

export const DAILY_LABEL = __( 'Day', 'tribe-events-calendar-pro' );
export const WEEKLY_LABEL = __( 'Week', 'tribe-events-calendar-pro' );
export const MONTHLY_LABEL = __( 'Month', 'tribe-events-calendar-pro' );
export const YEARLY_LABEL = __( 'Year', 'tribe-events-calendar-pro' );

export const DAILY_LABEL_PLURAL = __( 'Days', 'tribe-events-calendar-pro' );
export const WEEKLY_LABEL_PLURAL = __( 'Weeks', 'tribe-events-calendar-pro' );
export const MONTHLY_LABEL_PLURAL = __( 'Months', 'tribe-events-calendar-pro' );
export const YEARLY_LABEL_PLURAL = __( 'Years', 'tribe-events-calendar-pro' );

export const SINGLE_LABEL = __( 'Single Recurrence', 'tribe-events-calendar-pro' );

export const RECURRENCE_TYPES = [ DAILY, WEEKLY, MONTHLY, YEARLY, SINGLE ];

//
// ─── SERIES END TYPES ───────────────────────────────────────────────────────────
//

export const ON = 'on';
export const AFTER = 'after';
export const NEVER = 'never';

export const ON_LABEL = __( 'On', 'tribe-events-calendar-pro' );
export const AFTER_LABEL = __( 'After', 'tribe-events-calendar-pro' );
export const NEVER_LABEL = __( 'Never', 'tribe-events-calendar-pro' );

export const DATE = 'date';
export const COUNT = 'count';

//
// ─── DAYS OF THE WEEK ───────────────────────────────────────────────────────────
//

export const SUNDAY = 'sunday';
export const MONDAY = 'monday';
export const TUESDAY = 'tuesday';
export const WEDNESDAY = 'wednesday';
export const THURSDAY = 'thursday';
export const FRIDAY = 'friday';
export const SATURDAY = 'saturday';

export const SUNDAY_LABEL = __( 'Sunday', 'tribe-events-calendar-pro' );
export const MONDAY_LABEL = __( 'Monday', 'tribe-events-calendar-pro' );
export const TUESDAY_LABEL = __( 'Tuesday', 'tribe-events-calendar-pro' );
export const WEDNESDAY_LABEL = __( 'Wednesday', 'tribe-events-calendar-pro' );
export const THURSDAY_LABEL = __( 'Thursday', 'tribe-events-calendar-pro' );
export const FRIDAY_LABEL = __( 'Friday', 'tribe-events-calendar-pro' );
export const SATURDAY_LABEL = __( 'Saturday', 'tribe-events-calendar-pro' );

export const SUNDAY_ABBR = __( 'S', 'tribe-events-calendar-pro' );
export const MONDAY_ABBR = __( 'M', 'tribe-events-calendar-pro' );
export const TUESDAY_ABBR = __( 'T', 'tribe-events-calendar-pro' );
export const WEDNESDAY_ABBR = __( 'W', 'tribe-events-calendar-pro' );
export const THURSDAY_ABBR = __( 'T', 'tribe-events-calendar-pro' );
export const FRIDAY_ABBR = __( 'F', 'tribe-events-calendar-pro' );
export const SATURDAY_ABBR = __( 'S', 'tribe-events-calendar-pro' );

export const SUNDAY_CHECKED = 'sundayChecked';
export const MONDAY_CHECKED = 'mondayChecked';
export const TUESDAY_CHECKED = 'tuesdayChecked';
export const WEDNESDAY_CHECKED = 'wednesdayChecked';
export const THURSDAY_CHECKED = 'thursdayChecked';
export const FRIDAY_CHECKED = 'fridayChecked';
export const SATURDAY_CHECKED = 'saturdayChecked';

export const DAYS_OF_THE_WEEK_PROP_KEYS = [
	SUNDAY_CHECKED,
	MONDAY_CHECKED,
	TUESDAY_CHECKED,
	WEDNESDAY_CHECKED,
	THURSDAY_CHECKED,
	FRIDAY_CHECKED,
	SATURDAY_CHECKED,
];

export const DAYS_OF_THE_WEEK_MAPPING_TO_STATE = {
	[ MONDAY ]: 1,
	[ TUESDAY ]: 2,
	[ WEDNESDAY ]: 3,
	[ THURSDAY ]: 4,
	[ FRIDAY ]: 5,
	[ SATURDAY ]: 6,
	[ SUNDAY ]: 7,
};

export const DAYS_OF_THE_WEEK_MAPPING_FROM_STATE = invert( DAYS_OF_THE_WEEK_MAPPING_TO_STATE );

export const DAYS_OF_THE_WEEK_PROP_KEY_MAPPING_FROM_STATE = {
	1: MONDAY_CHECKED,
	2: TUESDAY_CHECKED,
	3: WEDNESDAY_CHECKED,
	4: THURSDAY_CHECKED,
	5: FRIDAY_CHECKED,
	6: SATURDAY_CHECKED,
	7: SUNDAY_CHECKED,
};

//
// ─── DAYS OF THE MONTH ──────────────────────────────────────────────────────────
//

// returns an array from 1 - 31
export const DAYS_OF_THE_MONTH = Array( 31 ).fill().map( ( _, index ) => index + 1 );

export const DAY = 'day';
export const DAY_LABEL = __( 'Day', 'tribe-events-calendar-pro' );

//
// ─── WEEKS OF THE MONTH ─────────────────────────────────────────────────────────
//

export const FIRST = 'first';
export const SECOND = 'second';
export const THIRD = 'third';
export const FOURTH = 'fourth';
export const FIFTH = 'fifth';
export const LAST = 'last';

export const FIRST_LABEL = __( 'First', 'tribe-events-calendar-pro' );
export const SECOND_LABEL = __( 'Second', 'tribe-events-calendar-pro' );
export const THIRD_LABEL = __( 'Third', 'tribe-events-calendar-pro' );
export const FOURTH_LABEL = __( 'Fourth', 'tribe-events-calendar-pro' );
export const FIFTH_LABEL = __( 'Fifth', 'tribe-events-calendar-pro' );
export const LAST_LABEL = __( 'Last', 'tribe-events-calendar-pro' );

export const WEEKS_OF_THE_MONTH = [ FIRST, SECOND, THIRD, FOURTH, FIFTH, LAST ];
export const WEEK_NUM_MAPPING_TO_WEEKS_OF_THE_MONTH = {
	1: FIRST,
	2: SECOND,
	3: THIRD,
	4: FOURTH,
	5: FIFTH,
};

//
// ─── MONTHS OF THE YEAR ─────────────────────────────────────────────────────────
//

export const JANUARY = 'january';
export const FEBRUARY = 'february';
export const MARCH = 'march';
export const APRIL = 'april';
export const MAY = 'may';
export const JUNE = 'june';
export const JULY = 'july';
export const AUGUST = 'august';
export const SEPTEMBER = 'september';
export const OCTOBER = 'october';
export const NOVEMBER = 'november';
export const DECEMBER = 'december';

export const JANUARY_LABEL = __( 'January', 'tribe-events-calendar-pro' );
export const FEBRUARY_LABEL = __( 'February', 'tribe-events-calendar-pro' );
export const MARCH_LABEL = __( 'March', 'tribe-events-calendar-pro' );
export const APRIL_LABEL = __( 'April', 'tribe-events-calendar-pro' );
export const MAY_LABEL = __( 'May', 'tribe-events-calendar-pro' );
export const JUNE_LABEL = __( 'June', 'tribe-events-calendar-pro' );
export const JULY_LABEL = __( 'July', 'tribe-events-calendar-pro' );
export const AUGUST_LABEL = __( 'August', 'tribe-events-calendar-pro' );
export const SEPTEMBER_LABEL = __( 'September', 'tribe-events-calendar-pro' );
export const OCTOBER_LABEL = __( 'October', 'tribe-events-calendar-pro' );
export const NOVEMBER_LABEL = __( 'November', 'tribe-events-calendar-pro' );
export const DECEMBER_LABEL = __( 'December', 'tribe-events-calendar-pro' );

export const JANUARY_ABBR = __( 'Jan', 'tribe-events-calendar-pro' );
export const FEBRUARY_ABBR = __( 'Feb', 'tribe-events-calendar-pro' );
export const MARCH_ABBR = __( 'Mar', 'tribe-events-calendar-pro' );
export const APRIL_ABBR = __( 'Apr', 'tribe-events-calendar-pro' );
export const MAY_ABBR = __( 'May', 'tribe-events-calendar-pro' );
export const JUNE_ABBR = __( 'Jun', 'tribe-events-calendar-pro' );
export const JULY_ABBR = __( 'Jul', 'tribe-events-calendar-pro' );
export const AUGUST_ABBR = __( 'Aug', 'tribe-events-calendar-pro' );
export const SEPTEMBER_ABBR = __( 'Sep', 'tribe-events-calendar-pro' );
export const OCTOBER_ABBR = __( 'Oct', 'tribe-events-calendar-pro' );
export const NOVEMBER_ABBR = __( 'Nov', 'tribe-events-calendar-pro' );
export const DECEMBER_ABBR = __( 'Dec', 'tribe-events-calendar-pro' );

export const MONTHS_OF_THE_YEAR_MAPPING_TO_STATE = {
	[ JANUARY ]: 1,
	[ FEBRUARY ]: 2,
	[ MARCH ]: 3,
	[ APRIL ]: 4,
	[ MAY ]: 5,
	[ JUNE ]: 6,
	[ JULY ]: 7,
	[ AUGUST ]: 8,
	[ SEPTEMBER ]: 9,
	[ OCTOBER ]: 10,
	[ NOVEMBER ]: 11,
	[ DECEMBER ]: 12,
};

export const MONTHS_OF_THE_YEAR_MAPPING_FROM_STATE = invert( MONTHS_OF_THE_YEAR_MAPPING_TO_STATE );

//
// ─── RECURRING MULTI DAY ────────────────────────────────────────────────────────
//

export const NEXT_DAY = 'next_day';
export const SECOND_DAY = 'second_day';
export const THIRD_DAY = 'third_day';
export const FOURTH_DAY = 'fourth_day';
export const FIFTH_DAY = 'fifth_day';
export const SIXTH_DAY = 'sixth_day';
export const SEVENTH_DAY = 'seventh_day';

export const NEXT_DAY_LABEL = __( 'Next day', 'tribe-events-calendar-pro' );
export const SECOND_DAY_LABEL = __( '2nd day', 'tribe-events-calendar-pro' );
export const THIRD_DAY_LABEL = __( '3rd day', 'tribe-events-calendar-pro' );
export const FOURTH_DAY_LABEL = __( '4th day', 'tribe-events-calendar-pro' );
export const FIFTH_DAY_LABEL = __( '5th day', 'tribe-events-calendar-pro' );
export const SIXTH_DAY_LABEL = __( '6th day', 'tribe-events-calendar-pro' );
export const SEVENTH_DAY_LABEL = __( '7th day', 'tribe-events-calendar-pro' );
