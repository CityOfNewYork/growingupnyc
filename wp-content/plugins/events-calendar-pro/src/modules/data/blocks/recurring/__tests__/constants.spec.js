/**
 * Internal Dependencies
 */
import * as constants from '../constants';

describe( 'Recurrence constants', () => {
	describe( 'Recurrence types', () => {
		test( 'daily', () => {
			expect( constants.DAILY ).toEqual( 'daily' );
		} );

		test( 'weekly', () => {
			expect( constants.WEEKLY ).toEqual( 'weekly' );
		} );

		test( 'monthly', () => {
			expect( constants.MONTHLY ).toEqual( 'monthly' );
		} );

		test( 'yearly', () => {
			expect( constants.YEARLY ).toEqual( 'yearly' );
		} );

		test( 'single', () => {
			expect( constants.SINGLE ).toEqual( 'single' );
		} );

		test( 'day label', () => {
			expect( constants.DAILY_LABEL ).toEqual( 'Day' );
		} );

		test( 'week label', () => {
			expect( constants.WEEKLY_LABEL ).toEqual( 'Week' );
		} );

		test( 'month label', () => {
			expect( constants.MONTHLY_LABEL ).toEqual( 'Month' );
		} );

		test( 'year label', () => {
			expect( constants.YEARLY_LABEL ).toEqual( 'Year' );
		} );

		test( 'single label', () => {
			expect( constants.SINGLE_LABEL ).toEqual( 'Single Recurrence' );
		} );

		test( 'recurrence types', () => {
			expect( constants.RECURRENCE_TYPES ).toEqual( [ 'daily', 'weekly', 'monthly', 'yearly', 'single' ] );
		} );
	} );

	describe( 'Series ends types', () => {
		test( 'on', () => {
			expect( constants.ON ).toEqual( 'on' );
		} );

		test( 'after', () => {
			expect( constants.AFTER ).toEqual( 'after' );
		} );

		test( 'never', () => {
			expect( constants.NEVER ).toEqual( 'never' );
		} );

		test( 'on label', () => {
			expect( constants.ON_LABEL ).toEqual( 'On' );
		} );

		test( 'after label', () => {
			expect( constants.AFTER_LABEL ).toEqual( 'After' );
		} );

		test( 'never label', () => {
			expect( constants.NEVER_LABEL ).toEqual( 'Never' );
		} );

		test( 'date', () => {
			expect( constants.DATE ).toEqual( 'date' );
		} );

		test( 'count', () => {
			expect( constants.COUNT ).toEqual( 'count' );
		} );
	} );

	describe( 'Days of the week', () => {
		test( 'sunday', () => {
			expect( constants.SUNDAY ).toEqual( 'sunday' );
		} );

		test( 'monday', () => {
			expect( constants.MONDAY ).toEqual( 'monday' );
		} );

		test( 'tuesday', () => {
			expect( constants.TUESDAY ).toEqual( 'tuesday' );
		} );

		test( 'wednesday', () => {
			expect( constants.WEDNESDAY ).toEqual( 'wednesday' );
		} );

		test( 'thursday', () => {
			expect( constants.THURSDAY ).toEqual( 'thursday' );
		} );

		test( 'friday', () => {
			expect( constants.FRIDAY ).toEqual( 'friday' );
		} );

		test( 'saturday', () => {
			expect( constants.SATURDAY ).toEqual( 'saturday' );
		} );

		test( 'sunday label', () => {
			expect( constants.SUNDAY_LABEL ).toEqual( 'Sunday' );
		} );

		test( 'monday label', () => {
			expect( constants.MONDAY_LABEL ).toEqual( 'Monday' );
		} );

		test( 'tuesday label', () => {
			expect( constants.TUESDAY_LABEL ).toEqual( 'Tuesday' );
		} );

		test( 'wednesday label', () => {
			expect( constants.WEDNESDAY_LABEL ).toEqual( 'Wednesday' );
		} );

		test( 'thursday label', () => {
			expect( constants.THURSDAY_LABEL ).toEqual( 'Thursday' );
		} );

		test( 'friday label', () => {
			expect( constants.FRIDAY_LABEL ).toEqual( 'Friday' );
		} );

		test( 'saturday label', () => {
			expect( constants.SATURDAY_LABEL ).toEqual( 'Saturday' );
		} );

		test( 'sunday abbreviation', () => {
			expect( constants.SUNDAY_ABBR ).toEqual( 'S' );
		} );

		test( 'monday abbreviation', () => {
			expect( constants.MONDAY_ABBR ).toEqual( 'M' );
		} );

		test( 'tuesday abbreviation', () => {
			expect( constants.TUESDAY_ABBR ).toEqual( 'T' );
		} );

		test( 'wednesday abbreviation', () => {
			expect( constants.WEDNESDAY_ABBR ).toEqual( 'W' );
		} );

		test( 'thursday abbreviation', () => {
			expect( constants.THURSDAY_ABBR ).toEqual( 'T' );
		} );

		test( 'friday abbreviation', () => {
			expect( constants.FRIDAY_ABBR ).toEqual( 'F' );
		} );

		test( 'saturday abbreviation', () => {
			expect( constants.SATURDAY_ABBR ).toEqual( 'S' );
		} );

		test( 'sunday checked', () => {
			expect( constants.SUNDAY_CHECKED ).toEqual( 'sundayChecked' );
		} );

		test( 'monday checked', () => {
			expect( constants.MONDAY_CHECKED ).toEqual( 'mondayChecked' );
		} );

		test( 'tuesday checked', () => {
			expect( constants.TUESDAY_CHECKED ).toEqual( 'tuesdayChecked' );
		} );

		test( 'wednesday checked', () => {
			expect( constants.WEDNESDAY_CHECKED ).toEqual( 'wednesdayChecked' );
		} );

		test( 'thursday checked', () => {
			expect( constants.THURSDAY_CHECKED ).toEqual( 'thursdayChecked' );
		} );

		test( 'friday checked', () => {
			expect( constants.FRIDAY_CHECKED ).toEqual( 'fridayChecked' );
		} );

		test( 'saturday checked', () => {
			expect( constants.SATURDAY_CHECKED ).toEqual( 'saturdayChecked' );
		} );

		test( 'days of the week prop keys', () => {
			expect( constants.DAYS_OF_THE_WEEK_PROP_KEYS ).toMatchSnapshot();
		} );

		test( 'days of the week mapping to state', () => {
			expect( constants.DAYS_OF_THE_WEEK_MAPPING_TO_STATE ).toMatchSnapshot();
		} );

		test( 'days of the week mapping from state', () => {
			expect( constants.DAYS_OF_THE_WEEK_MAPPING_FROM_STATE ).toMatchSnapshot();
		} );

		test( 'days of the week prop key mapping from state', () => {
			expect( constants.DAYS_OF_THE_WEEK_PROP_KEY_MAPPING_FROM_STATE ).toMatchSnapshot();
		} );
	} );

	describe( 'Days of the month', () => {
		test( 'days of the month', () => {
			expect( constants.DAYS_OF_THE_MONTH ).toMatchSnapshot();
		} );

		test( 'day', () => {
			expect( constants.DAY ).toEqual( 'day' );
		} );

		test( 'day label', () => {
			expect( constants.DAY_LABEL ).toEqual( 'Day' );
		} );
	} );

	describe( 'Weeks of the month', () => {
		test( 'first', () => {
			expect( constants.FIRST ).toEqual( 'first' );
		} );

		test( 'second', () => {
			expect( constants.SECOND ).toEqual( 'second' );
		} );

		test( 'third', () => {
			expect( constants.THIRD ).toEqual( 'third' );
		} );

		test( 'fourth', () => {
			expect( constants.FOURTH ).toEqual( 'fourth' );
		} );

		test( 'fifth', () => {
			expect( constants.FIFTH ).toEqual( 'fifth' );
		} );

		test( 'last', () => {
			expect( constants.LAST ).toEqual( 'last' );
		} );

		test( 'first label', () => {
			expect( constants.FIRST_LABEL ).toEqual( 'First' );
		} );

		test( 'second label', () => {
			expect( constants.SECOND_LABEL ).toEqual( 'Second' );
		} );

		test( 'third label', () => {
			expect( constants.THIRD_LABEL ).toEqual( 'Third' );
		} );

		test( 'fourth label', () => {
			expect( constants.FOURTH_LABEL ).toEqual( 'Fourth' );
		} );

		test( 'fifth label', () => {
			expect( constants.FIFTH_LABEL ).toEqual( 'Fifth' );
		} );

		test( 'last label', () => {
			expect( constants.LAST_LABEL ).toEqual( 'Last' );
		} );

		test( 'weeks of the month', () => {
			expect( constants.WEEKS_OF_THE_MONTH ).toEqual( [ 'first', 'second', 'third', 'fourth', 'fifth', 'last' ] );
		} );

		test( 'week num mapping to weeks of the month', () => {
			expect( constants.WEEK_NUM_MAPPING_TO_WEEKS_OF_THE_MONTH ).toMatchSnapshot();
		} );
	} );

	describe( 'Months of the year', () => {
		test( 'january', () => {
			expect( constants.JANUARY ).toEqual( 'january' );
		} );

		test( 'february', () => {
			expect( constants.FEBRUARY ).toEqual( 'february' );
		} );

		test( 'march', () => {
			expect( constants.MARCH ).toEqual( 'march' );
		} );

		test( 'april', () => {
			expect( constants.APRIL ).toEqual( 'april' );
		} );

		test( 'may', () => {
			expect( constants.MAY ).toEqual( 'may' );
		} );

		test( 'june', () => {
			expect( constants.JUNE ).toEqual( 'june' );
		} );

		test( 'july', () => {
			expect( constants.JULY ).toEqual( 'july' );
		} );

		test( 'august', () => {
			expect( constants.AUGUST ).toEqual( 'august' );
		} );

		test( 'september', () => {
			expect( constants.SEPTEMBER ).toEqual( 'september' );
		} );

		test( 'october', () => {
			expect( constants.OCTOBER ).toEqual( 'october' );
		} );

		test( 'november', () => {
			expect( constants.NOVEMBER ).toEqual( 'november' );
		} );

		test( 'december', () => {
			expect( constants.DECEMBER ).toEqual( 'december' );
		} );

		test( 'january label', () => {
			expect( constants.JANUARY_LABEL ).toEqual( 'January' );
		} );

		test( 'february label', () => {
			expect( constants.FEBRUARY_LABEL ).toEqual( 'February' );
		} );

		test( 'march label', () => {
			expect( constants.MARCH_LABEL ).toEqual( 'March' );
		} );

		test( 'april label', () => {
			expect( constants.APRIL_LABEL ).toEqual( 'April' );
		} );

		test( 'may label', () => {
			expect( constants.MAY_LABEL ).toEqual( 'May' );
		} );

		test( 'june label', () => {
			expect( constants.JUNE_LABEL ).toEqual( 'June' );
		} );

		test( 'july label', () => {
			expect( constants.JULY_LABEL ).toEqual( 'July' );
		} );

		test( 'august label', () => {
			expect( constants.AUGUST_LABEL ).toEqual( 'August' );
		} );

		test( 'september label', () => {
			expect( constants.SEPTEMBER_LABEL ).toEqual( 'September' );
		} );

		test( 'october label', () => {
			expect( constants.OCTOBER_LABEL ).toEqual( 'October' );
		} );

		test( 'november label', () => {
			expect( constants.NOVEMBER_LABEL ).toEqual( 'November' );
		} );

		test( 'december label', () => {
			expect( constants.DECEMBER_LABEL ).toEqual( 'December' );
		} );

		test( 'january abbreviation', () => {
			expect( constants.JANUARY_ABBR ).toEqual( 'Jan' );
		} );

		test( 'february abbreviation', () => {
			expect( constants.FEBRUARY_ABBR ).toEqual( 'Feb' );
		} );

		test( 'march abbreviation', () => {
			expect( constants.MARCH_ABBR ).toEqual( 'Mar' );
		} );

		test( 'april abbreviation', () => {
			expect( constants.APRIL_ABBR ).toEqual( 'Apr' );
		} );

		test( 'may abbreviation', () => {
			expect( constants.MAY_ABBR ).toEqual( 'May' );
		} );

		test( 'june abbreviation', () => {
			expect( constants.JUNE_ABBR ).toEqual( 'Jun' );
		} );

		test( 'july abbreviation', () => {
			expect( constants.JULY_ABBR ).toEqual( 'Jul' );
		} );

		test( 'august abbreviation', () => {
			expect( constants.AUGUST_ABBR ).toEqual( 'Aug' );
		} );

		test( 'september abbreviation', () => {
			expect( constants.SEPTEMBER_ABBR ).toEqual( 'Sep' );
		} );

		test( 'october abbreviation', () => {
			expect( constants.OCTOBER_ABBR ).toEqual( 'Oct' );
		} );

		test( 'november abbreviation', () => {
			expect( constants.NOVEMBER_ABBR ).toEqual( 'Nov' );
		} );

		test( 'december abbreviation', () => {
			expect( constants.DECEMBER_ABBR ).toEqual( 'Dec' )
		} );

		test( 'months of the year mapping to state', () => {
			expect( constants.MONTHS_OF_THE_YEAR_MAPPING_TO_STATE ).toMatchSnapshot();
		} );

		test( 'months of the year mapping from state', () => {
			expect( constants.MONTHS_OF_THE_YEAR_MAPPING_FROM_STATE ).toMatchSnapshot();
		} );
	} );

	describe( 'recurring multi day', () => {
		test( 'next day', () => {
			expect( constants.NEXT_DAY ).toEqual( 'next_day' );
		} );

		test( 'second day', () => {
			expect( constants.SECOND_DAY ).toEqual( 'second_day' );
		} );

		test( 'third day', () => {
			expect( constants.THIRD_DAY ).toEqual( 'third_day' );
		} );

		test( 'fourth day', () => {
			expect( constants.FOURTH_DAY ).toEqual( 'fourth_day' );
		} );

		test( 'fifth day', () => {
			expect( constants.FIFTH_DAY ).toEqual( 'fifth_day' );
		} );

		test( 'sixth day', () => {
			expect( constants.SIXTH_DAY ).toEqual( 'sixth_day' );
		} );

		test( 'seventh day', () => {
			expect( constants.SEVENTH_DAY ).toEqual( 'seventh_day' );
		} );

		test( 'next day label', () => {
			expect( constants.NEXT_DAY_LABEL ).toEqual( 'Next day' );
		} );

		test( 'second day label', () => {
			expect( constants.SECOND_DAY_LABEL ).toEqual( '2nd day' );
		} );

		test( 'third day label', () => {
			expect( constants.THIRD_DAY_LABEL ).toEqual( '3rd day' );
		} );

		test( 'fourth day label', () => {
			expect( constants.FOURTH_DAY_LABEL ).toEqual( '4th day' );
		} );

		test( 'fifth day label', () => {
			expect( constants.FIFTH_DAY_LABEL ).toEqual( '5th day' );
		} );

		test( 'sixth day label', () => {
			expect( constants.SIXTH_DAY_LABEL ).toEqual( '6th day' );
		} );

		test( 'seventh day label', () => {
			expect( constants.SEVENTH_DAY_LABEL ).toEqual( '7th day' );
		} );
	} );
} );
