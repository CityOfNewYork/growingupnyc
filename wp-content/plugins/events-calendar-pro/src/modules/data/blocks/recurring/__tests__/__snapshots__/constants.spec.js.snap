// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recurrence constants Days of the month days of the month 1`] = `
Array [
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
  11,
  12,
  13,
  14,
  15,
  16,
  17,
  18,
  19,
  20,
  21,
  22,
  23,
  24,
  25,
  26,
  27,
  28,
  29,
  30,
  31,
]
`;

exports[`Recurrence constants Days of the week days of the week mapping from state 1`] = `
Object {
  "1": "monday",
  "2": "tuesday",
  "3": "wednesday",
  "4": "thursday",
  "5": "friday",
  "6": "saturday",
  "7": "sunday",
}
`;

exports[`Recurrence constants Days of the week days of the week mapping to state 1`] = `
Object {
  "friday": 5,
  "monday": 1,
  "saturday": 6,
  "sunday": 7,
  "thursday": 4,
  "tuesday": 2,
  "wednesday": 3,
}
`;

exports[`Recurrence constants Days of the week days of the week prop key mapping from state 1`] = `
Object {
  "1": "mondayChecked",
  "2": "tuesdayChecked",
  "3": "wednesdayChecked",
  "4": "thursdayChecked",
  "5": "fridayChecked",
  "6": "saturdayChecked",
  "7": "sundayChecked",
}
`;

exports[`Recurrence constants Days of the week days of the week prop keys 1`] = `
Array [
  "sundayChecked",
  "mondayChecked",
  "tuesdayChecked",
  "wednesdayChecked",
  "thursdayChecked",
  "fridayChecked",
  "saturdayChecked",
]
`;

exports[`Recurrence constants Months of the year months of the year mapping from state 1`] = `
Object {
  "1": "january",
  "10": "october",
  "11": "november",
  "12": "december",
  "2": "february",
  "3": "march",
  "4": "april",
  "5": "may",
  "6": "june",
  "7": "july",
  "8": "august",
  "9": "september",
}
`;

exports[`Recurrence constants Months of the year months of the year mapping to state 1`] = `
Object {
  "april": 4,
  "august": 8,
  "december": 12,
  "february": 2,
  "january": 1,
  "july": 7,
  "june": 6,
  "march": 3,
  "may": 5,
  "november": 11,
  "october": 10,
  "september": 9,
}
`;

exports[`Recurrence constants Weeks of the month week num mapping to weeks of the month 1`] = `
Object {
  "1": "first",
  "2": "second",
  "3": "third",
  "4": "fourth",
  "5": "fifth",
}
`;
